src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -667,7 +678,7 @@ private void ThreadProc()
     {
         while (!_cts.IsCancellationRequested)
```

Comment:
```suggestion
        while (!_cts.Token.WaitHandle.WaitOne(1_000 / 30))
```

Less code and avoids the last `DisplayNodes` call when the waithandle is signaled.